src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -593,44 +593,64 @@ findNeighborRanks( std::vector< vtkBoundingBox > boundingBoxes )
   return neighbors;
 }
 
-vtkSmartPointer< vtkDataSet >
-redistributeMesh( vtkDataSet & loadedMesh,
-                  MPI_Comm const comm,
-                  PartitionMethod const method,
-                  int const partitionRefinement,
-                  int const useGlobalIds )
+vtkSmartPointer< vtkDataSet > manageGlobalIds( vtkSmartPointer< vtkDataSet > mesh, int useGlobalIds )
 {
-  GEOS_MARK_FUNCTION;
+  auto hasGlobalIds = []( vtkSmartPointer< vtkDataSet > m ) -> bool
+  {
+    return m->GetPointData()->GetGlobalIds() != nullptr && m->GetCellData()->GetGlobalIds() != nullptr;
+  };
 
-  // Generate global IDs for vertices and cells, if needed
-  vtkSmartPointer< vtkDataSet > mesh;
-  bool globalIdsAvailable = loadedMesh.GetPointData()->GetGlobalIds() != nullptr
-                            && loadedMesh.GetCellData()->GetGlobalIds() != nullptr;
+  {
+    // Add global ids on the fly if needed
+    int const me = hasGlobalIds( mesh );
+    int everyone;
+    MpiWrapper::allReduce( &me, &everyone, 1, MPI_MAX, MPI_COMM_GEOSX );
+
+    if( everyone and not me )
```

Comment:
Would this be ok?
```suggestion
    int const anyone = MpiWrapper::max( me );

    if( anyone && !me )
```
(I don't object to use of `and`, `not`, `or` etc. keywords in c++ in general, this is just not the style GEOS uses everywhere else, even in this PR)

Reply:
Yes, I will fix it

Reply:
Yes, we'll fix this.

For the context, @francoishamon did cherry pick this in one of my branches.
So I'll fix it in my branch to make my merge easier, if you do not mind!